src/ai/backend/manager/models/rbac_models/role.py

```diff
@@ -72,3 +76,81 @@ class RoleRow(Base):
         back_populates="role_row",
         primaryjoin="RoleRow.id == foreign(ObjectPermissionRow.role_id)",
     )
+
+    def to_data(self) -> RoleData:
+        return RoleData(
+            id=self.id,
+            name=self.name,
+            status=self.status,
+            created_at=self.created_at,
+            updated_at=self.updated_at,
+            deleted_at=self.deleted_at,
+            description=self.description,
+        )
+
+    def to_data_with_permissions(
+        self, active_permission_only: bool = True
+    ) -> RoleDataWithPermissions:
+        if active_permission_only:
+            scope_permissions = [
+                sp.to_data_with_entity()
+                for sp in self.scope_permission_rows
+                if sp.status == PermissionStatus.ACTIVE
+            ]
+            object_permissions = [
+                op.to_data()
+                for op in self.object_permission_rows
+                if op.status == PermissionStatus.ACTIVE
+            ]
+        else:
+            scope_permissions = [sp.to_data_with_entity() for sp in self.scope_permission_rows]
+            object_permissions = [op.to_data() for op in self.object_permission_rows]
+        return RoleDataWithPermissions(
+            id=self.id,
+            name=self.name,
+            status=self.status,
+            scope_permissions=scope_permissions,
+            object_permissions=object_permissions,
+            created_at=self.created_at,
+            updated_at=self.updated_at,
+            deleted_at=self.deleted_at,
+            description=self.description,
+        )
+
+    @classmethod
+    def _get_value[T](cls, getter: Callable[..., T]) -> tuple[Optional[T], bool]:
+        try:
+            return getter(), True
+        except ValueError:
+            return None, False
```

Comment:
I'm not sure if this method is needed. Shouldn't we just use the methods of OptionalState?